### ISSUE TITLE
is "links with identical names serve equivalent purpose" (b20e66) still needed?

### ISSUE THREAD

**Issue by Jym77** [act-rules/act-rules.github.io#962]
This stem from the discussion starting at https://github.com/act-rules/act-rules.github.io/pull/768#discussion_r330184189

There's 2 SC about link purpose: 2.4.4 (level A, "link purpose (in context)") and 2.4.9 (level AAA, "link purpose (link only)").
We currently have one published rule for both ([Link has accessible name](https://act-rules.github.io/rules/c487ae)) which only checks the presence of an accessible name, not its accuracy; and one for 2.4.9 ([Links with identical accessible names have equivalent purpose](https://act-rules.github.io/rules/b20e66)). @carlosapaduarte is working on #555 for 2.4.4 which checks the accuracy of individual links.

There are also proposal/discussions to adapt [Links with identical accessible names have equivalent purpose](https://act-rules.github.io/rules/b20e66) for 2.4.4 by adding context (#693) and to adapt #555 for 2.4.9 by removing context (not really needed since it's AAA, but almost "free" once #555 is done).

---
Now, the thing is that the SC is concerned about the accuracy of the description of each individual link and doesn't care about "set of links with the same name". Obviously, if every individual link name is accurate, then links with the same name are OK.

So, once we have "link is descriptive" (with/without context), we won't really need "identical links serve equivalent purpose" (with/without context).

But, "identical links serve equivalent purpose" is also a nice heuristic for "link is descriptive", especially for semi-automated tools. Rather than asking whether each link is descriptive, we can start by selecting out, say, 3 out of 100 links on the page and ask whether it is normal that they have same name but different destination. Not needed for coverage (because less precise than "link is descriptive") but can be a "nice to have" for tools, and we already have it (at least for 2.4.9 which is also where applicability is super-easy to automate because there is no context)…

---
Anyway, what now?

1. We can decide that both versions (2.4.4 and 2.4.9) of both rules are interesting and keep working on all, aka nothing changes.
2. We can decide that "link is descriptive" supersedes "identical links". In that case, we should kill #693 as soon as #555 is merged (or even immediately), and put the current [Links with identical accessible names have equivalent purpose](https://act-rules.github.io/rules/b20e66) in "low maintenance" mode until we have time to write "link is descriptive" for 2.4.9 (this probably means finished #768 and #959 that are currently open and close to done, but not doing any more work on it).
3. We can decide that the future is to only have "link is descriptive" and commit to it. That would kill #693, #768 and #959 immediately (we stop working on "identical links"). But we're committing to write "link is descriptive (link only)" soon-ish to replace the current [Links with identical accessible names have equivalent purpose](https://act-rules.github.io/rules/b20e66).
4. Other ideas?

**Comment by WilcoFiers** [comment]
I don't think having both rules is a problem, for the same reason that when we have an "image alt is descriptive" rule it makes "image filename is descriptive". Having overlap between rules is fine.

**Comment by Jym77** [comment]
"links with identical names" are relatively easy to (semi-)automate while "link is descriptive" rules are essentially manual, so it is worth to keep both.

**Comment by Jym77** [comment]
Following up on @DagfinnRomen remark during WAI-tool call (and keeping examples from the previous discussion)…

```html
Joe is drinking a <a href=#cup-of-coffee>cup of joe</a>) in the 
<a href=#cup-of-a-guy-named-joe>cup of Joe</a>.
```
could be an example of links with identical name ("cup of joe") (almost identical names, depends if capitalisation matters) but serving different purpose and still both are fairly descriptive (thus 2.4.9 is not broken).

That would pass 2.4.9 and "link is descriptive (link only)" but fail "link with identical names serve equivalent purpose".

That would also pass 2.4.4, "link in context is descriptive", and be inapplicable for "links with identical names **and context** serve equivalent purpose".

**Comment by Jym77** [comment]
Note that while the rule does not map to SC 2.4.9, it may still map to another SC, or at least be considered best practice and kept as such…

**Comment by Jym77** [comment]
Note: given that the "links with identical name" rules are actually not mapping to 2.4.4/2.4.9 (see #968 for examples) and can probably only be kept as best practice, they are also not actually subset of the "link is descriptive" rules making this whole issue not really problematic.

The question is whether we want to keep this as best practice or not, but this is related to #968 rather than this issue.

**Comment by Jym77** [comment]
_De facto_ superseded by #968. Closing.
